https://github.com/orblivion/iiab/actions/runs/24413611959/job/71316798194

At the end of Build Image:

```
Ansible quit before photograph. Exiting with failure.
Error: Process completed with exit code 1.
```

Any idea @chapmanjacobd ? I don't see any obvious actual cause.

I looked for "error" above in the logs and found this, along with a few other thing that fail the same way. I'm guessing unrelated, since it didn't stop the installation right away:

```
TASK [yarn : debug] ************************************************************
Warning: : Encountered 1 template error.
error 1 - 'yarn_installed' is undefined
Origin: /opt/iiab/iiab/roles/yarn/tasks/main.yml:26:10

24     var: yarn_enabled
25 - debug:
26     var: yarn_installed
            ^ column 10

ok: [127.0.0.1] => {
    "yarn_installed": "<< error 1 - 'yarn_installed' is undefined >>"
}
```

--- a/roles/maps/defaults/main.yml
+++ b/roles/maps/defaults/main.yml
@@ -98,10 +98,10 @@ maps_dot_black_naturalearth6_symlink_name: "naturalearth6-NE2_HR_SR_W_DR-WEBP.pm
 # Mostly colors, topography, etc.
 maps_dot_black_naturalearth6_tiles:
   # For actual users
-  full: "naturalearth6-NE2_HR_SR_W_DR-WEBP.{{ maps_slow_data_date }}.z00-06.pmtiles"
+  full: "naturalearth6-NE2_HR_SR_W_DR-WEBP.{{ maps_slow_data_date }}.z00-z06.pmtiles"
 
   # FOR TESTING ONLY
-  ci: "naturalearth6-NE2_HR_SR_W_DR-WEBP.{{ maps_slow_data_date }}.z00-04.pmtiles"
+  ci: "naturalearth6-NE2_HR_SR_W_DR-WEBP.{{ maps_slow_data_date }}.z00-z04.pmtiles"
 
 # Two different symlinks because the front end could requset openstreetmap or naturalearth
 maps_dot_black_osm_symlink_name: openstreetmap-openmaptiles.pmtiles
@@ -120,12 +120,12 @@ maps_dot_black_vector_tiles:
   # (old) "medium res" osm, up to zoom level 9 (original file has 14).
   # (TODO does this include colors and topography? Or is it used along with naturalearth6 above in most styles?)
   # maps_vector_quality = "osm-z9"
-  osm-z9: "openstreetmap-openmaptiles.{{ maps_vector_data_date }}.z00-09.pmtiles"
+  osm-z9: "openstreetmap-openmaptiles.{{ maps_vector_data_date }}.z00-z09.pmtiles"
 
   # (new) "medium res" osm, up to zoom level 11 (original file has 14).
   # (TODO does this include colors and topography? Or is it used along with naturalearth6 above in most styles?)
   # maps_vector_quality = "osm-z11"
-  osm-z11: "openstreetmap-openmaptiles.{{ maps_vector_data_date }}.z00-11.pmtiles"
+  osm-z11: "openstreetmap-openmaptiles.{{ maps_vector_data_date }}.z00-z11.pmtiles"
 
   # "low res" - mostly borders, rivers, country names, large roads.
   # maps_vector_quality = "ne"
@@ -135,7 +135,7 @@ maps_dot_black_vector_tiles:
   # FOR TESTING ONLY
   # "medium res" osm, up to zoom level 1 (original file has 14).
   # maps_vector_quality = "osm-z1"
-  osm-z1: "openstreetmap-openmaptiles.{{ maps_vector_data_date }}.z00-01.pmtiles"
+  osm-z1: "openstreetmap-openmaptiles.{{ maps_vector_data_date }}.z00-z01.pmtiles"
 
 maps_vector_max_zoom: 14
 
@@ -144,19 +144,19 @@ maps_dot_black_satellite_symlink_name: s2maps-sentinel2-2023.pmtiles
 maps_dot_black_satellite_tiles:
   # Low quality satellite, up to zoom level 7 (original file has 13)
   # maps_satellite_zoom = 7
-  7: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-07.pmtiles"
+  7: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-z07.pmtiles"
 
   # Moderately high quality satellite, up to zoom level 9 (original file has 13)
   # maps_satellite_zoom = 9
-  9: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-09.pmtiles"
+  9: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-z09.pmtiles"
 
   # Pretty high quality satellite, up to zoom level 11 (original file has 13)
   # maps_satellite_zoom = 11
-  11: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-11.pmtiles"
+  11: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-z11.pmtiles"
 
   # Pretty high quality satellite, up to zoom level 12 (original file has 13)
   # maps_satellite_zoom = 12
-  12: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-12.pmtiles"
+  12: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-z12.pmtiles"
 
   # Highest available quality satellite, up to zoom level 13
   # maps_satellite_zoom = 13
@@ -165,7 +165,7 @@ maps_dot_black_satellite_tiles:
   # FOR TESTING ONLY
   # Super-low quality satellite, up to zoom level 4 (original file has 13)
   # maps_satellite_zoom = 4
-  4: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-04.pmtiles"
+  4: "s2maps-sentinel2-2023.{{ maps_slow_data_date }}.z00-z04.pmtiles"
 
 maps_satellite_max_zoom: 13
 
@@ -179,13 +179,13 @@ maps_dot_black_terrain_symlink_name: terrarium-z0-z10.pmtiles
 maps_dot_black_terrain_tiles:
   # Low quality terrain, up to zoom level 7 (original file has 10)
   # maps_terrain_zoom = 7
-  7: "terrarium.{{ maps_slow_data_date }}.z00-07.pmtiles"
+  7: "terrarium.{{ maps_slow_data_date }}.z00-z07.pmtiles"
 
   # maps_terrain_zoom = 8
-  8: "terrarium.{{ maps_slow_data_date }}.z00-08.pmtiles"
+  8: "terrarium.{{ maps_slow_data_date }}.z00-z08.pmtiles"
 
   # maps_terrain_zoom = 9
-  9: "terrarium.{{ maps_slow_data_date }}.z00-09.pmtiles"
+  9: "terrarium.{{ maps_slow_data_date }}.z00-z09.pmtiles"
 
   # maps_terrain_zoom = 10
   # (This is the highest quality that maps.black offers in pmtiles format. They
@@ -195,8 +195,8 @@ maps_dot_black_terrain_tiles:
   # NOT RECOMMENDED. VERY LOW QUALITY terrain, up to zoom level 5 or 6.
   # We will probably end up getting rid of these, but we would like to test
   # them out to be sure.
-  5: "terrarium.{{ maps_slow_data_date }}.z00-05.pmtiles"
-  6: "terrarium.{{ maps_slow_data_date }}.z00-06.pmtiles"
+  5: "terrarium.{{ maps_slow_data_date }}.z00-z05.pmtiles"
+  6: "terrarium.{{ maps_slow_data_date }}.z00-z06.pmtiles"
 
 maps_terrain_max_zoom: 10
 

--- a/roles/maps/templates/index.html.j2
+++ b/roles/maps/templates/index.html.j2
@@ -143,6 +143,8 @@
                 },
 
                 getPmtilesName: (baseName) => {
+                    const pmtilesDates = downloadedFQRegions[activeFQRegionName].dates
+
                     // baseName and fname won't include ".pmtiles". ".pmtiles" gets added by the calling function.
                     if (!activeFQRegionName) {
                         fname = baseName
@@ -151,13 +153,13 @@
                         // terrarium actually starts with "terrarium-z0-z10".
                         // We chop off the z0-z10 to avoid confusing since we
                         // make lower zoom versions.
-                        fname = `terrarium.${MAPS_SLOW_DATA_DATE}.full-region.` + activeFQRegionName
+                        fname = `terrarium.${pmtilesDates.terrain}.full-region.` + activeFQRegionName
                         console.log("pmtiles file name for full quality region", fname)
                     } else if (baseName.startsWith("openstreetmap")) {
-                        fname = baseName + `.${MAPS_VECTOR_DATA_DATE}.full-region.` + activeFQRegionName
+                        fname = baseName + `.${pmtilesDates.vector}.full-region.` + activeFQRegionName
                         console.log("pmtiles file name for full quality region", fname)
                     } else if (baseName.startsWith("s2maps")) {
-                        fname = baseName + `.${MAPS_SLOW_DATA_DATE}.full-region.` + activeFQRegionName
+                        fname = baseName + `.${pmtilesDates.satellite}.full-region.` + activeFQRegionName
                         console.log("pmtiles file name for full quality region", fname)
                     } else {
                         fname = baseName
@@ -248,7 +250,7 @@
                 for (regionName_ in downloadedFQRegions) {
                     const regionName = regionName_ // fix variable capturing in callbacks below
                     const regionId = "region-" + regionName
-                    const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[regionName]
+                    const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[regionName].bbox
 
                     if (mb.map.getSource(regionId)) {
                         // Looks like we already ran this. Quit without setting a timeout.
@@ -354,7 +356,7 @@
                         if (regionName === activeFQRegionName) {
                             return
                         }
-                        const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[regionName]
+                        const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[regionName].bbox
 
                         // First we kick off a zoom animation to a view focusing on the region.
                         mb.map.fitBounds([[min_lon, min_lat], [max_lon, max_lat]], {})
@@ -520,7 +522,7 @@
                     this._container.className = 'maplibregl-ctrl maplibregl-ctrl-group';
                     this._container.style = 'cursor: pointer;'
                     this._container.onclick = () => {
-                        const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[activeFQRegionName]
+                        const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[activeFQRegionName].bbox
 
                         // First we kick off a zoom animation to a view focusing on the region, zoomed out a bit more than when we entered it.
                         mb.map.setMaxBounds() // untether us from the bounds first so we can zoom out further
@@ -568,7 +570,7 @@
                     this._container.className = 'maplibregl-ctrl maplibregl-ctrl-group';
                     this._container.style = 'cursor: pointer;'
                     this._container.onclick = () => {
-                        const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[activeFQRegionName]
+                        const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[activeFQRegionName].bbox
 
                         // Zoom out to view the region so that the delete popup is well placed
                         mb.map.fitBounds([[min_lon, min_lat], [max_lon, max_lat]], {maxDuration: 2000, speed: 5})
@@ -1116,7 +1118,7 @@
 
                         // We are looking at a full quality region. Let's prevent
                         // the user from straying far from it so they don't get lost.
-                        const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[activeFQRegionName]
+                        const [min_lon, min_lat, max_lon, max_lat] = downloadedFQRegions[activeFQRegionName].bbox
                         mb.map.setMaxBounds([[min_lon - 0.5, min_lat - 0.5], [max_lon + 0.5, max_lat + 0.5]], {})
                     }
 

--- a/roles/maps/templates/index.html.j2
+++ b/roles/maps/templates/index.html.j2
@@ -33,7 +33,7 @@
              *    configs   *
              ****************/
 
-            const MAPS_DATA_DATE = "{{ maps_data_date }}"
+            const MAPS_VECTOR_DATA_DATE = "{{ maps_vector_data_date }}"
             const MAPS_SLOW_DATA_DATE = "{{ maps_slow_data_date }}"
             const MAPS_VECTOR_QUALITY = "{{ maps_vector_quality }}"
             const MAPS_VECTOR_IS_OSM = MAPS_VECTOR_QUALITY.includes("osm")
@@ -154,7 +154,7 @@
                         fname = `terrarium.${MAPS_SLOW_DATA_DATE}.full-region.` + activeFQRegionName
                         console.log("pmtiles file name for full quality region", fname)
                     } else if (baseName.startsWith("openstreetmap")) {
-                        fname = baseName + `.${MAPS_DATA_DATE}.full-region.` + activeFQRegionName
+                        fname = baseName + `.${MAPS_VECTOR_DATA_DATE}.full-region.` + activeFQRegionName
                         console.log("pmtiles file name for full quality region", fname)
                     } else if (baseName.startsWith("s2maps")) {
                         fname = baseName + `.${MAPS_SLOW_DATA_DATE}.full-region.` + activeFQRegionName

--- a/roles/maps/templates/tile-extract.py.j2
+++ b/roles/maps/templates/tile-extract.py.j2
@@ -1,6 +1,7 @@
 #!/usr/bin/python3
 
-import collections, json, glob, os, re, shutil, string, subprocess, sys
+import json, glob, os, re, shutil, string, subprocess, sys
+from collections import defaultdict
 
 HOSTING_DIR = "{{ maps_serve_path }}"
 EXTRACTS_JSON = os.path.join(HOSTING_DIR, "{{ tile_extract.info_file }}")
@@ -190,13 +191,24 @@ def check_for_overlaps(new_box):
     return True
 
 def update_extracts_json():
-    region_extracts = {}
-    region_extract_types = collections.defaultdict(set)
+    # region_extracts[name] = {
+    #   "bbox": coordinates,
+    #   "dates": {
+    #     "terrain": terrain_date,
+    #     "vector": vector_date,
+    #     "satellite": satellite_date,
+    #   },
+    # }
+    region_extracts = defaultdict(
+        lambda: dict(dates=defaultdict(dict)),
+    )
+    # region_extract_types[name] = {extract_type_1, extract_type_2, etc}
+    region_extract_types = defaultdict(set)
 
     for region_extract in glob.glob(os.path.join(HOSTING_DIR, "*.full-region.*.pmtiles")):
 
         match (os.path.basename(region_extract).split('.')):
-            case [region_extract_type, _date, "full-region", name, "pmtiles"]:
+            case [region_extract_type, date, "full-region", name, "pmtiles"]:
                 pass
             case _:
                 raise Exception("Oops can't parse pmtiles file name:", region_extract)
@@ -205,9 +217,17 @@ def update_extracts_json():
         coordinates = [float(coord) for coord in json.loads(output)["bounds"]]
 
         # Add coordinates unless it's for some reason inconsistent between satellite, vector, terrain.
-        if name in region_extracts and region_extracts[name] != coordinates:
+        if name in region_extracts and region_extracts[name]['bbox'] != coordinates:
             raise Exception("Inconsistent coordinates between different region extracts for", name)
-        region_extracts[name] = coordinates
+        if not re.match('\d{4}-\d{2}-\d{2}', date):
+            raise Exception(f"Invalid date for {name}: {date}")
+        region_extracts[name]["bbox"] = coordinates
+        region_type_for_date = {
+            's2maps-sentinel2-2023': 'satellite',
+            'terrain': 'terrarium',
+            'vector': 'openstreetmap-openmaptiles',
+        }[region_extract_type]
+        region_extracts[name]["dates"][region_type_for_date] = date
         region_extract_types[name].add(region_extract_type)
 
     for name in region_extract_types:

